node/mock-task.ts

```diff
@@ -397,21 +368,33 @@ export interface MatchOptions {
     nonegate?: boolean;
     flipNegate?: boolean;
 }
-export function match(list: string[], pattern: string, options?: MatchOptions): string[];
-export function match(list: string[], patterns: string[], options?: MatchOptions): string[];
-export function match(list: string[], pattern: any, options?: MatchOptions): string[] {
-    return mock.getResponse('match', pattern) || [];
+
+export function match(list: string[], patterns: string[] | string, patternRoot?: string, options?: MatchOptions): string[] {
+    let responseKey: string = typeof patterns == 'object' ? (patterns as string[]).join('\n') : patterns as string;
+    let result: string[] = mock.getResponse('match', responseKey, module.exports.debug);
```

Comment:
maybe not mock at all, since it's not touch disk, and user has full control of all inputs.